Mathlib/NumberTheory/LSeries/RiemannZeta.lean

```diff
@@ -0,0 +1,246 @@
+
+## Outline of proofs:
+
+These results are mostly special cases of more general results for even Hurwitz zeta functions
+proved in `Mathlib.NumberTheory.LSeries.HurwitzZetaEven`.
+-/
+
+
+open MeasureTheory Set Filter Asymptotics TopologicalSpace Real Asymptotics
+  Classical HurwitzZeta
+
+open Complex hiding exp norm_eq_abs abs_of_nonneg abs_two continuous_exp
+
+open scoped Topology Real Nat
+
+noncomputable section
+
+/-!
+## Definition of the completed Riemann zeta
+-/
+
+/-- The completed Riemann zeta function with its poles removed, `Λ(s) + 1 / s - 1 / (s - 1)`. -/
+def completedRiemannZeta₀ (s : ℂ) : ℂ := completedHurwitzZetaEven₀ 0 s
+#align riemann_completed_zeta₀ completedRiemannZeta₀
+
+/-- The completed Riemann zeta function, `Λ(s)`, which satisfies
+`Λ(s) = π ^ (-s / 2) Γ(s / 2) ζ(s)` (up to a minor correction at `s = 0`). -/
+def completedRiemannZeta (s : ℂ) : ℂ := completedHurwitzZetaEven 0 s
+#align riemann_completed_zeta completedRiemannZeta
+
+lemma HurwitzZeta.completedHurwitzZetaEven_zero (s : ℂ) :
+    completedHurwitzZetaEven 0 s = completedRiemannZeta s := by rfl
+
+lemma HurwitzZeta.completedHurwitzZetaEven₀_zero (s : ℂ) :
+    completedHurwitzZetaEven₀ 0 s = completedRiemannZeta₀ s := by rfl
+
+lemma HurwitzZeta.completedCosZeta_zero (s : ℂ) :
+    completedCosZeta 0 s = completedRiemannZeta s := by
+  rw [completedRiemannZeta, completedHurwitzZetaEven, completedCosZeta, hurwitzEvenFEPair_zero_symm]
+
+lemma HurwitzZeta.completedCosZeta₀_zero (s : ℂ) :
+    completedCosZeta₀ 0 s = completedRiemannZeta₀ s := by
+  rw [completedRiemannZeta₀,
+    completedHurwitzZetaEven₀, completedCosZeta₀, hurwitzEvenFEPair_zero_symm]
+
+lemma completedRiemannZeta_eq (s : ℂ) :
+    completedRiemannZeta s = completedRiemannZeta₀ s - 1 / s - 1 / (1 - s) := by
+  simp_rw [completedRiemannZeta, completedRiemannZeta₀, completedHurwitzZetaEven_eq, if_true]
+
+/-- The modified completed Riemann zeta function `Λ(s) + 1 / s + 1 / (1 - s)` is entire. -/
+theorem differentiable_completedZeta₀ : Differentiable ℂ completedRiemannZeta₀ :=
+  differentiable_completedHurwitzZetaEven₀ 0
+#align differentiable_completed_zeta₀ differentiable_completedZeta₀
+
+/-- The completed Riemann zeta function `Λ(s)` is differentiable away from `s = 0` and `s = 1`. -/
+theorem differentiableAt_completedZeta {s : ℂ} (hs : s ≠ 0) (hs' : s ≠ 1) :
+    DifferentiableAt ℂ completedRiemannZeta s :=
+  differentiableAt_completedHurwitzZetaEven 0 (Or.inl hs) hs'
+
+/-- Riemann zeta functional equation, formulated for `Λ₀`: for any complex `s` we have
+`Λ₀(1 - s) = Λ₀ s`. -/
+theorem completedRiemannZeta₀_one_sub (s : ℂ) :
+    completedRiemannZeta₀ (1 - s) = completedRiemannZeta₀ s := by
+  rw [← completedHurwitzZetaEven₀_zero, ← completedCosZeta₀_zero, completedHurwitzZetaEven₀_one_sub]
+#align riemann_completed_zeta₀_one_sub completedRiemannZeta₀_one_sub
+
+@[deprecated completedRiemannZeta₀_one_sub (since := "2024-05-27")]
```

Comment:
Why is this deprecated? Is there a replacement elsewhere?

Reply:
This is just a backward-compatibility measure: I renamed some declarations, and there's a convention that when you rename `oldName` to `newName` you add `@[deprecated] alias oldName newName` so anyone still using `oldName` in their code will get a polite message saying "oldName is deprecated, use newName", rather than everything just stopping working. (Cf. Sebastien's recent post in the PNT+ thread, about the experience of bumping the mathlib version used in the PNT+ repo.)

As for _why_ the declarations were renamed: originally we had `riemannCompletedZeta`, ported from `riemann_completed_zeta` in Mathlib3. On rewriting this stuff, I didn't like the look of `hurwitzCompletedZeta` -- it's a completion of Hurwitz' zeta function, not Hurwitz' completion of a zeta function (while both readings work for Riemann!) so went for `completedHurwitzZeta`, and changed the Riemann zeta for consistency with that.

Reply:
(PS. Evidently you already know this, since you use the same mechanism in your `AddChar` PR; and I learned from your code there that one does not need to specify the new name in the deprecation.)

Reply:
My point is that you deprecated the *new* names. This was confusing me.

Reply:
> My point is that you deprecated the new names. This was confusing me.

No, I don't think I did, unless I'm grotesquely misunderstanding how the deprecation mechanism works. The old definitions were `riemannCompletedXXX`, they have been replaced by `completedRiemanXXX`, and I am adding back `riemannCompletedXXX` as a deprecated alias for the new declaration.

Reply:
My understanding is that when you write `@[deprecated <name>]`, then this deprecates `<name>` (and `<name>` was `completedRiemann...`.

Reply:
> My understanding is that when you write `@[deprecated <name>]`, then this deprecates `<name>` (and `<name>` was `completedRiemann...`.

No, the `<name>` is the replacement.

Reply:
Thanks for clarifying (and sorry for the noise!).